src/c_api/c_api.cc

```diff
@@ -248,11 +250,21 @@ void CustomFComputeDispatcher(const std::string op_name,
   // get actual cudaStream_t out of mxnet gpu stream and pass to lib_api.h
   void *cuda_stream = nullptr;
 #if MXNET_USE_CUDA
-  if (inputs[0].ctx().dev_mask() == Context::kGPU) {
+  if (inputs.size() > 0 && inputs[0].ctx().dev_mask() == Context::kGPU) {
```

Comment:
while you're making this change, its possible an operator might have zero inputs and some outputs (ie. generator operator, like random), should we also check for outputs if inputs is zero like:
```
if ((inputs.size() > 0 && inputs[0].ctx().dev_mask() == Context::kGPU) ||
    (outputs.size() > 0 && inputs[0].ctx().dev_mask() == Context::kGPU)) {
```

Reply:
I think we just check for output